crates/jstzd/tests/octez_client_test.rs

```diff
@@ -351,3 +358,185 @@ async fn send_rollup_inbox_message() {
+            &proof.commitment,
+            &format!("0x{}", &proof.proof),
+            Some(0.1),
+        )
+        .await
+        .unwrap();
+
+    tokio::time::timeout(
+        tokio::time::Duration::from_secs(5),
+        octez_client.wait_for(&op, Some(&block), None),
+    )
+    .await
+    .expect("wait_for should complete soon enough")
+    .expect("wait_for should be able to find the operation");
+
+    let operation_kind =
+        get_operation_kind(&octez_node.rpc_endpoint().to_string(), &block, &op)
+            .await
+            .unwrap();
+    assert_eq!(operation_kind, "smart_rollup_execute_outbox_message");
+
+    let _ = rollup.kill().await;
+    let _ = baker.kill().await;
+    let _ = octez_node.kill().await;
+}
+
+async fn set_up_parameters_for_outbox_message(
+    rollup_address: &str,
+    contract_address: &str,
+    installer_path: &PathBuf,
+    contract_path: &PathBuf,
+) -> ProtocolParameter {
+    let kernel = String::from_utf8(
+        tokio::fs::read(&installer_path)
+            .await
+            .unwrap_or_else(|e| panic!("failed to read installer file: {:?}", e)),
+    )
+    .unwrap();
+    let contract_json = serde_json::from_slice(
+        &tokio::fs::read(&contract_path)
+            .await
+            .unwrap_or_else(|e| panic!("failed to read contract file: {:?}", e)),
+    )
+    .unwrap();
+    let params = ProtocolParameterBuilder::new()
+        .set_bootstrap_smart_rollups([BootstrapSmartRollup::new(
+            rollup_address,
+            SmartRollupPvmKind::Wasm,
+            &kernel,
+            serde_json::json!({"prim": "bytes"}),
+        )
+        .unwrap()])
+        .set_bootstrap_contracts([BootstrapContract::new(
+            contract_json,
+            1_000_000,
+            Some(contract_address),
+        )
+        .unwrap()])
+        .set_source_path(
+            Path::new(std::env!("CARGO_MANIFEST_DIR"))
+                .join("tests/sandbox-params.json")
+                .to_str()
+                .unwrap(),
+        )
+        .build()
+        .unwrap();
+
+    let mut content = tokio::fs::read_to_string(params.parameter_file().path())
+        .await
+        .unwrap();
+    let mut value: serde_json::Value = serde_json::from_str(&content).unwrap();
+
+    // overwriting these config values so that outbox messages get cemented sooner
+    value.as_object_mut().unwrap().insert(
+        "smart_rollup_challenge_window_in_blocks".to_owned(),
+        serde_json::json!(8),
+    );
+    value.as_object_mut().unwrap().insert(
+        "smart_rollup_commitment_period_in_blocks".to_owned(),
+        serde_json::json!(8),
+    );
+
+    content = serde_json::to_string(&value).unwrap();
+    tokio::fs::File::create(params.parameter_file().path())
+        .await
+        .unwrap()
+        .write_all(content.as_bytes())
+        .await
+        .unwrap();
+
+    params
+}
+
+async fn wait_for_outbox_proof(rollup_rpc_endpoint: &str) -> anyhow::Result<OutputProof> {
```

Comment:
Note: we should turn this into an `OctezRollupClient` function

Reply:
For now I think it's better to keep this function here because there are too many assumptions based on the test setup in this implementation. We can polish and move it when we need it.

Reply:
Can you add a ticket to track it?

Reply:
https://linear.app/tezos/issue/JSTZ-206/polish-wait-for-outbox-proof-for-general-use